catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -263,6 +277,8 @@ public void updateCallback(Map<String, Object> properties) {
         setAttributeOverrides(Arrays.asList(parameterArray));
       }
 
+      destroy(0);
```

Comment:
❗️ You are updating properties above that you will use when setting up routes later. This is done from a different thread and therefore should be also protected by your lock.